core/templates/dev/head/forms/formBuilder.js

```diff
@@ -906,10 +913,16 @@ oppia.filter('isFloat', [function() {
       return undefined;
     }
 
-    if (viewValue !== '' && viewValue !== '-' && FLOAT_REGEXP.test(viewValue)) {
-      return parseFloat(viewValue.replace(',', '.'));
-    }
-    return undefined;
+    if (viewValue !== '' && FLOAT_REGEXP.test(viewValue)) {
+      if (viewValue.slice(-1) === '%'){
+
+        // This is a percentage, so the input needs to be divided by 100.
+        return parseFloat(viewValue.replace(',', '.')) / 100;
+      } else{
```

Comment:
minor nit: add single space before '{'

(Follow [Google JavaScript style guide](http://google-styleguide.googlecode.com/svn/trunk/javascriptguide.xml#Code_formatting) when in doubt.)